minkindr/include/kindr/minimal/implementation/rotation-quaternion-inl.h

```diff
@@ -407,8 +409,9 @@ RotationQuaternionTemplate<Scalar>::operator*(
 
   // check if the multiplication has resulted in the quaternion no longer being
   // approximately normalized.
-  if (std::abs(result.squaredNorm() - static_cast<Scalar>(1.0)) >
-      EPS<Scalar>::normalization_value()) {
```

Comment:
Not familiar with jet types or how they're used -- are you templating Scalar for it then? 
Looks like there are specializations of abs (just not std::abs) for jet types -- https://github.com/kashif/ceres-solver/blob/master/include/ceres/jet.h#L401-L405 -- could you then have a different specialization of this function for jet types instead?